interfaces/builtin/x11.go

```diff
@@ -60,14 +120,67 @@ socket AF_NETLINK - NETLINK_KOBJECT_UEVENT
 bind
 `
 
+type x11Interface struct{}
+
+func (iface *x11Interface) Name() string {
+	return "x11"
+}
+
+func (iface *x11Interface) StaticInfo() interfaces.StaticInfo {
+	return interfaces.StaticInfo{
+		Summary:              x11Summary,
+		ImplicitOnClassic:    true,
+		BaseDeclarationSlots: x11BaseDeclarationSlots,
+	}
+}
+
+func (iface *x11Interface) SecCompConnectedPlug(spec *seccomp.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	spec.AddSnippet(x11ConnectedPlugSecComp)
+	return nil
+}
+
+func (iface *x11Interface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	spec.AddSnippet(x11ConnectedPlugAppArmor)
+	return nil
+}
+
+func (iface *x11Interface) AppArmorConnectedSlot(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	if !release.OnClassic {
+		// TODO: figure out how to share X11 socket with other snaps
```

Comment:
You want a rule like this:
```
unix (connect, receive, send)
     type=stream
     addr="@/tmp/.X11-unix/X[0-9]*"
     peer=(label=###PLUG_SECURITY_TAGS###),
```

maliit.go has an example of this.

Reply:
Done. Yes I was unaware that abstract sockets aren't file backed, and thus not affected by mount namespaces. This completely solves my socket sharing trouble